src/common/libidset/idset.h

```diff
@@ -142,6 +142,14 @@ struct idset *idset_intersect (const struct idset *a, const struct idset *b);
  */
 bool idset_has_intersection (const struct idset *a, const struct idset *b);
 
+/* Convenience interfaces for using idset as an "integer allocator".
+ * idset_free() doesn't check for double frees, but idset_free_check() does.
+ * It returns -1 and sets errno to EEXIST when that happens.
+ */
+int idset_alloc (struct idset *idset, unsigned int *val);
+void idset_free (struct idset *idset, unsigned int val);
```

Comment:
It appears these methods will fail if the idset was not created with `IDSET_FLAG_INITFULL`, which by name or comment is not obviously related. Perhaps a comment here, or rename the flag to something more obvious `IDSET_FLAG_ALLOCATOR`?

Reply:
Well since INITFULL does not necessarily imply the allocator, maybe a comment would do?  I did say something in the man page but the header could be explicit too.

Reply:
A comment is fine - since it is critical, in the header file would be nice.